### PR TITLE
hw patterns

### PR DESCRIPTION
**Pull request by Taniann**

### CODE REVIEW

File: src/main/java/ua/mateacademy/tania/patterns/proxy/ProxyFile.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    private RealFile realFile;
+    private String fileName;
+
+    public ProxyFile(String fileName){
+        this.fileName = fileName;
+    }
+
+    @Override
+    public void execute() {
+        if(realFile == null){
```
Comment (DadizhaO) [inline src/main/java/ua/mateacademy/tania/patterns/proxy/ProxyFile.java:13]: Think equals better there

**Review by DadizhaO — APPROVED** [review]
good

File: src/main/java/ua/mateacademy/tania/patterns/abstractfactory/AbstractFactoryApp.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package ua.mateacademy.tania.patterns.abstractfactory;
+
+import ua.mateacademy.tania.patterns.abstractfactory.polish.PolishFactory;
+
+public class AbstractFactoryApp {
+    public static void main(String[] args) {
+        Factory factory = new PolishFactory();
```
Comment (SerhiiPasko) [inline src/main/java/ua/mateacademy/tania/patterns/abstractfactory/AbstractFactoryApp.java]: You use concretic factory|class. It is hardcode. Read https://www.baeldung.com/java-abstract-factory-pattern

File: src/main/java/ua/mateacademy/tania/patterns/adapter/AdapterPatternApp.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package ua.mateacademy.tania.patterns.adapter;
+
+/**
+ * Created by Таня on 06.08.2018.
+ */
+public class AdapterPatternApp {
+    public static void main(String[] args) {
+        DVIConnector dviConnector = new DVIConnectorAdapterToSVGA();
```
Comment (SerhiiPasko) [inline src/main/java/ua/mateacademy/tania/patterns/adapter/AdapterPatternApp.java:8]: You have some instance of SVGA
And must use it as DVI

Comment (SerhiiPasko) [reply src/main/java/ua/mateacademy/tania/patterns/adapter/AdapterPatternApp.java:8]: SVGAConnector ->convert to DVIConnectorAdapterToSVGA ->use as DVIConnector

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

File: src/main/java/ua/mateacademy/tania/patterns/adapter/AdapterPatternApp.java (hunk 3 above)
Comment (Taniann) [reply src/main/java/ua/mateacademy/tania/patterns/adapter/AdapterPatternApp.java:8]: > SVGAConnector ->convert to DVIConnectorAdapterToSVGA ->use as DVIConnector

yes, i actually do this. In class lesson we've used composition. in my hw I use inharitance

Comment (SerhiiPasko) [reply src/main/java/ua/mateacademy/tania/patterns/adapter/AdapterPatternApp.java:8]: This steps illogical. You before have some object and must use it as another. But you directly create some between without start predictions

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comment

**Pull request merged**